app/models/race_edition.rb

```diff
@@ -1,16 +1,19 @@
 class RaceEdition < ActiveRecord::Base
+  extend FriendlyId
+
   belongs_to :race
   has_many :race_entries, dependent: :destroy
   has_many :racers, through: :race_entries
 
   accepts_nested_attributes_for :racers
+  friendly_id :name, use: :slugged
 
   validates :race_id, presence: true
   validates :date, presence: true
   validates :entry_fee, presence: true
+  validates_uniqueness_of :race_id, scope: :date
 
   def name
-    race.name + ' on ' + date.to_s
+    "#{race&.name} on #{date}"
```

Comment:
This is a stylistic but also a substantive change. If race&.name returns nil, string concatenation will fail, whereas string interpolation will convert it to an empty string. We need to handle the nil case to ensure all tests pass, including the test that validates that RaceEdition is invalid without a race_id.

Reply:
Sounds good